pointcloud/basic_octree_utils_test.go

```diff
@@ -327,3 +330,75 @@ func printBasicOctree(logger golog.Logger, bOct *BasicOctree, s string) {
 		printBasicOctree(logger, v, s+"-+-")
 	}
 }
+
+// Test the functionalities involved with converting a pointcloud into a basic octree.
+func TestBasicOctreeCollision(t *testing.T) {
+	startPC, err := makeFullPointCloudFromArtifact(
+		t,
+		"pointcloud/collision_pointcloud_0.pcd",
+		BasicType,
+	)
+	test.That(t, err, test.ShouldBeNil)
+
+	center := getCenterFromPcMetaData(startPC.MetaData())
+	maxSideLength := getMaxSideLengthFromPcMetaData(startPC.MetaData())
+
+	basicOct, err := NewBasicOctree(center, maxSideLength)
+	test.That(t, err, test.ShouldBeNil)
+
+	startPC.Iterate(0, 0, func(p r3.Vector, d Data) bool {
+		// Blue channel is used to determine probability in pcds produced by cartographer
+		_, _, blueProb := d.RGB255()
+		d.SetValue(int(blueProb))
+		if err = basicOct.Set(p, d); err != nil {
+			return false
+		}
+		return true
+	})
+
+	test.That(t, startPC.Size(), test.ShouldEqual, basicOct.Size())
+
+	// create a non-colliding obstacle far away from any octree point
+	far, err := spatialmath.NewBox(spatialmath.NewZeroPose(), r3.Vector{1, 2, 3}, "far")
```

Comment:
[opt] I'd suggest reorganizing to include the obstacle creation in the associated `t.Run()`. I would feel fine with either combining the two cases that use 'near' in one `t.Run()` or defining the 'near' obstacle twice. Up to you!

Reply:
Done